testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java

```diff
@@ -173,6 +177,8 @@ protected void insertTableRows(String tableName, String floatToDoubleType)
                         "  TIMESTAMP '2121-07-15 15:30:12.123', " +
                         "  TIMESTAMP '2121-07-15 15:30:12.123', " +
                         "  TIMESTAMP '2121-07-15 15:30:12.123', " +
+                        "  '2121', " +
+                        "  '2121-07-15 15:30:12.123', " +
```

Comment:
add cases with micros and nano precision, 
`.123499`
`.123500`
`.123501`
`.123499999`
`.123500000`
`.123500001`

Reply:
But these cases cannot be applied at the same time right ?

Reply:
i don't understand the question.
do you mean we're limited to just one test data point?

Reply:
The existing test we have is limited to run with a default precision - I'll try to add additional test coverage by having a dedicated table wit varying precision or we could try be setting the default precision to `NANO_SECONDS`

Reply:
good point. we should probably exercise this logic with all supported precisions
let's have a dedicated test